shared/modules/conversion.utils.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Currency Conversion Utility
+ * This utility function can be used for converting currency related values within metamask.
+ * The caller should be able to pass it a value, along with information about the value's
+ * numeric base, denomination and currency, and the desired numeric base, denomination and
+ * currency. It should return a single value.
+ *
+ * @param {(number | string | BN)} value - The value to convert.
+ * @param {object} [options] - Options to specify details of the conversion
+ * @param {string} [options.fromCurrency = EtherDenomination.ETH | 'USD'] - The currency of the passed value
+ * @param {string} [options.toCurrency = EtherDenomination.ETH | 'USD'] - The desired currency of the result
+ * @param {string} [options.fromNumericBase = 'hex' | 'dec' | 'BN'] - The numeric basic of the passed value.
+ * @param {string} [options.toNumericBase = 'hex' | 'dec' | 'BN'] - The desired numeric basic of the result.
+ * @param {string} [options.fromDenomination = EtherDenomination.WEI] - The denomination of the passed value
+ * @param {string} [options.numberOfDecimals] - The desired number of decimals in the result
+ * @param {string} [options.roundDown] - The desired number of decimals to round down to
+ * @param {number} [options.conversionRate] - The rate to use to make the fromCurrency -> toCurrency conversion
+ * @returns {(number | string | BN)}
+ *
+ * The utility passes value along with the options as a single object to the `converter` function.
+ * `converter` conditional modifies the supplied `value` property, depending
+ * on the accompanying options.
+ */
+
+import { BigNumber } from 'bignumber.js';
+
+import { addHexPrefix, BN } from 'ethereumjs-util';
+import { EtherDenomination } from '../constants/common';
+import { Numeric } from './Numeric';
+
+/**
+ * Defines the base type of numeric value
+ */
+export type NumericBase = 'hex' | 'dec' | undefined;
+
+/**
```

Comment:
This seems left over from the documentation for EthDenomination — should it be ✂️ ?

Reply:
Removed